tensorrt_llm/_torch/visual_gen/mapping.py

```
@@ -14,14 +14,17 @@
-from torch.distributed.device_mesh import init_device_mesh
+from torch.distributed.device_mesh import DeviceMesh, init_device_mesh
 
 from tensorrt_llm._torch.device_mesh import DeviceMeshTopologyImpl, SingleProcessGroup
 from tensorrt_llm.logger import logger
 from tensorrt_llm.mapping import Mapping
 
-_VALID_DIM_NAMES = frozenset({"cfg", "tp", "cp", "ulysses"})
-DEFAULT_DIM_ORDER = "cfg-tp-cp-ulysses"
+# Legacy mesh: single CP dimension (ring or unused placeholder size 1).
+_DEVICE_MESH_DIM_ORDER_LEGACY = "cfg-tp-cp-ulysses"
+# Attention2D mesh: split CP into a 2D tile (cp_row × cp_col) so row/col
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)